### PR TITLE
Issue/jperf 566 fix ci flakiness

### PR DESCRIPTION
**Pull request by dagguh**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  build:
+    environment:
+      TERM: dumb
+    machine: true
+    working_directory: ~/repo
+    steps:
+      - checkout
+      - restore_cache:
+          keys:
+            - v1-dependencies-{{ checksum "build.gradle.kts" }}
```
Comment (wyrzyk) [inline .circleci/config.yml]: YAML programmer 👍

Comment (dagguh) [reply .circleci/config.yml]: 😂

**Comment by dagguh** [comment]
Here's 4 runs of da9abac :
* https://circleci.com/gh/dagguh/docker-infrastructure/7
* https://circleci.com/gh/dagguh/docker-infrastructure/8
* https://circleci.com/gh/dagguh/docker-infrastructure/9
* https://circleci.com/gh/dagguh/docker-infrastructure/10

It still flakes. I could bump the timeout even further, but there is something fishy here. Looking in the stdout of the tests from the test reports in the artifacts we can look for the `org.apache.catalina.startup.Catalina.start Server startup in` line. We see two such lines, due to two test methods. One reports in 11-13 seconds and the other (docker-to-docker) is 170-280 seconds. And then yet again, one times out completely despite an 8 minute timeout.

I'll try to replace the log polling with GUI polling

**Comment by dagguh** [comment]
https://circleci.com/workflow-run/fa542720-2a09-40a1-ba5e-fc00fc432c4f
This flake is the second kind: it's unhappy with the admin fullname field.
This blocks me from merging https://github.com/atlassian/jira-actions/pull/29 and I have no idea how to fix it.

**Comment by dagguh** [comment]
This is still not working, I'll close and reopen when CircleCI goes green

**Comment by dagguh** [comment]
Turned out that the license setup takes around 60 seconds and our timeout for the next screen is also 60 seconds. I bumped the WD timeout and now [it's all green](https://circleci.com/workflow-run/3a4d7dd6-ac50-4c45-865a-1f550b9d2085).
The extra timeout means the CI runs for a little longer.
So this trades a minute (or two) of dev loop for reliability. In total, this should save a lot of time, due to no need for reruns.

File: src/main/kotlin/com/atlassian/performance/tools/dockerinfrastructure/jira/SetUpFromScratchAction.kt [hunk 2]
```diff
@@ -28,8 +29,23 @@ internal class SetUpFromScratchAction(
...
 
+    private fun waitForJira() {
+        driver.wait(
+            condition = ExpectedCondition {
+                it!!
+                return@ExpectedCondition try {
+                    it.navigate().to(uri.toURL())
+                    it.findElements(By.id("jira")).firstOrNull()
+                } catch (e: Exception) {
+                    null
```
Comment (wyrzyk) [inline src/main/kotlin/com/atlassian/performance/tools/dockerinfrastructure/jira/SetUpFromScratchAction.kt]: 😱

Comment (dagguh) [reply src/main/kotlin/com/atlassian/performance/tools/dockerinfrastructure/jira/SetUpFromScratchAction.kt]: That's the `ExpectedCondition` SPI

**Pull request merged**